Solution1/BusinessLayer/Services/WeatherService.cs

```diff
@@ -54,5 +57,51 @@ public async Task<ForecastWeatherDTO> GetForecastByCityNameAsync(string cityName
 
             return _mapper.Map<ForecastWeatherDTO>(forecast).FillCommentByTemp(); 
         }
+
+        public async Task<Dictionary<bool, List<WeatherResponseDTO>>> GetWeatherByArrayCityNameAsync(string arrayCityNames)
+        {
+            var cityNames = arrayCityNames.Split(',').Select(cityName => cityName.Trim());
+            
+            var timer = new Stopwatch();
+            timer.Start();
+
+            var listTasksRequest = cityNames.Select(async cityName =>
+            {
+                var weatherResponseDTO = new WeatherResponseDTO() { CityName = cityName };
+                try
+                {
+                    ValidationCityName(cityName);
+                    weatherResponseDTO.Temp = (await _weatherApiService.GetByCityNameAsync(cityName)).TemperatureValues.Temp;
+                    weatherResponseDTO.IsSuccessfulRequest = true;
+                }
+                catch (ValidationException ex)
+                {
+                    weatherResponseDTO.IsSuccessfulRequest = false;
+                    weatherResponseDTO.ErrorMessage = ex.Errors.FirstOrDefault().ErrorMessage;
+                }
+                catch (Exception ex)
+                {
+                    weatherResponseDTO.IsSuccessfulRequest = false;
+                    weatherResponseDTO.ErrorMessage = ex.Message;
+                }
+                weatherResponseDTO.LeadTime = timer.ElapsedMilliseconds;
+                return weatherResponseDTO;
+            });
+
+            var weatherResponses = await Task.WhenAll(listTasksRequest);
+            return weatherResponses.GroupBy(w => w.IsSuccessfulRequest).ToDictionary(k => k.Key, v => v.ToList());
```

Comment:
make to list only when it's really necessary

Reply:
+